### PR TITLE
TSDB: Add dimensions and timestamp to parse errors

### PR DESCRIPTION
**Pull request by nik9000**
This adds a list of dimensions and `@timestamp` to the parse errors we
make in tsdb if the `_tsid` or `@timestamp` are built.

### CODE REVIEW

**Review by csoulios — APPROVED** [review]
LGTM!

**Comment by nik9000** [comment]
This relates to #84957

File: server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java [hunk 1]
```diff
@@ -156,13 +156,25 @@ public void createField(DocumentParserContext context, BytesRef tsid) {
...
+        StringBuilder description = new StringBuilder("a time series document");
+        IndexableField tsidField = context.doc().getField(TimeSeriesIdFieldMapper.NAME);
+        if (tsidField != null) {
+            String tsid = TimeSeriesIdFieldMapper.decodeTsid(tsidField.binaryValue()).toString();
+            description.append(" with dimensions ").append(tsid);
+        }
+        IndexableField timestampField = context.doc().getField(DataStreamTimestampFieldMapper.DEFAULT_PATH);
+        if (timestampField != null) {
+            String timestamp = DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.formatMillis(timestampField.numericValue().longValue());
+            description.append(" at ").append(timestamp);
```
[the remark this answers is not in the document]
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java:176]: Makes sense to me.

File: server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java [hunk 2]
```diff
@@ -156,13 +156,25 @@ public void createField(DocumentParserContext context, BytesRef tsid) {
...
+         * time series dimensions and the timestamp! If we have those, then
+         * include them in the description. If not, all we know is
+         * "a time series document".
          */
-        return "a time series document";
+        StringBuilder description = new StringBuilder("a time series document");
+        IndexableField tsidField = context.doc().getField(TimeSeriesIdFieldMapper.NAME);
+        if (tsidField != null) {
+            String tsid = TimeSeriesIdFieldMapper.decodeTsid(tsidField.binaryValue()).toString();
+            description.append(" with dimensions ").append(tsid);
```
Comment (nik9000) [inline server/src/main/java/org/elasticsearch/index/mapper/TsidExtractingIdFieldMapper.java:171]: @henningandersen and I talked about limiting this to maybe 1k characters or something. I hate that it'd make debugging hard but it's nice defense against a *huge* bulk response.

**Review by romseygeek — APPROVED** [review]
LGTM

**Review by imotov — APPROVED** [review]
LGTM, Thanks!

**Comment by nik9000** [comment]
@elasticmachine test this please

**Comment by nik9000** [comment]
@elasticmachine test this please

**Pull request merged**